metadata-ingestion/src/datahub/ingestion/source/dbt/dbt_common.py

```diff
@@ -977,16 +981,20 @@ def create_platform_mces(
                     f"Skipping emission of node {node_datahub_urn} because node_type {node.node_type} is disabled"
                 )
                 continue
-            self.stale_entity_removal_handler.add_entity_to_state(
-                "dataset", node_datahub_urn
-            )
+            if not is_primary_source:
+                # We previously, erroneously added non-dbt nodes to the state object.
+                # This call ensures that we don't try to soft-delete them after an
+                # upgrade of acryl-datahub.
+                self.stale_entity_removal_handler.add_urn_to_skip(node_datahub_urn)
```

Comment:
qq - wouldn't this case already be covered by putting the is_primary_source into the workunit itself? 

confused as to why we need both

Reply:
we were previously calling the add_entity_to_state for some snowflake nodes that we weren't even emitting. because of that, using the is_primary piece isn't sufficient to avoid soft-deleting stuff that accidentally

Reply:
oh mannnn